packages/charts/chart-web-components/src/donut-chart/donut-chart.ts

```diff
@@ -0,0 +1,257 @@
+
+  private _arcs: SVGPathElement[] = [];
+  private _isRTL: boolean = false;
+  private _textInsideDonut?: SVGTextElement;
+
+  constructor() {
+    super();
+
+    this.elementInternals.role = 'region';
+  }
+
+  public handleLegendMouseoverAndFocus(legendTitle: string) {
+    if (this.isLegendSelected) {
+      return;
+    }
+
+    this.activeLegend = legendTitle;
+  }
+
+  public handleLegendMouseoutAndBlur() {
+    if (this.isLegendSelected) {
+      return;
+    }
+
+    this.activeLegend = '';
+  }
+
+  public handleLegendClick(legendTitle: string) {
+    if (this.isLegendSelected && this.activeLegend === legendTitle) {
+      this.activeLegend = '';
+      this.isLegendSelected = false;
+    } else {
+      this.activeLegend = legendTitle;
+      this.isLegendSelected = true;
+    }
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    validateChartProps(this.data, 'data');
+
+    this.data.chartData.forEach((dataPoint, index) => {
+      if (dataPoint.color) {
+        dataPoint.color = getColorFromToken(dataPoint.color);
+      } else {
+        dataPoint.color = getNextColor(index);
+      }
+    });
+
+    this.legends = this._getLegends();
+    this._isRTL = getRTL(this);
+    this.elementInternals.ariaLabel =
+      this.data.chartTitle || `Donut chart with ${this.data.chartData.length} segments.`;
+
+    this._render();
+  }
+
+  private _render() {
+    const pie = d3Pie<ChartDataPoint>()
+      .value(d => d.data)
+      .padAngle(0.02);
+    const arc = d3Arc<PieArcDatum<ChartDataPoint>>()
+      .innerRadius(this.innerRadius)
+      .outerRadius((Math.min(this.height, this.width) - 20) / 2);
+
+    pie(this.data.chartData).forEach(arcDatum => {
+      const arcGroup = document.createElementNS(SVG_NAMESPACE_URI, 'g');
+      this.group.appendChild(arcGroup);
+
+      const pathOutline = document.createElementNS(SVG_NAMESPACE_URI, 'path');
+      arcGroup.appendChild(pathOutline);
+      pathOutline.classList.add('arc-outline');
+      pathOutline.setAttribute('d', arc(arcDatum)!);
+
+      const path = document.createElementNS(SVG_NAMESPACE_URI, 'path');
+      arcGroup.appendChild(path);
+      this._arcs.push(path);
+      path.classList.add('arc');
+      path.setAttribute('d', arc(arcDatum)!);
+      path.setAttribute('fill', arcDatum.data.color!);
+      path.setAttribute('data-id', arcDatum.data.legend);
+      path.setAttribute('tabindex', '0');
+      path.setAttribute('aria-label', `${arcDatum.data.legend}, ${arcDatum.data.data}.`);
+      path.setAttribute('role', 'img');
+
+      path.addEventListener('mouseover', event => {
+        if (this.activeLegend !== '' && this.activeLegend !== arcDatum.data.legend) {
+          return;
+        }
+
+        const bounds = this.getBoundingClientRect();
+
+        this.tooltipProps = {
+          isVisible: true,
+          legend: arcDatum.data.legend,
+          yValue: `${arcDatum.data.data}`,
```

Comment:
Does this need to be interpolated as a string?

Reply:
Yes. This is needed. We need to add a formatter prop to format the number into a proper SI string representation.